### PR TITLE
Make images inline elements

### PR DESCRIPTION
**Pull request by bensyverson**
## Summary
tl;dr: This one-line PR allows images to create new paragraphs, by treating them like links, which are also inline elements.

## Overview
This PR addresses an issue with the way that Ink currently handles images. Essentially, when an image is encountered outside of an existing paragraph fragment, Ink treats it as a fragment at the same level as `Heading`, `HTML`, `Blockquote`,  `CodeBlock`, `HorizontalLine` or `Paragraph`.

This prevents a bare image in its own paragraph from being surrounded by `<p></p>`, and probably causes some other undesirable behavior. Images should be treated the same as links, which are very similar and **do** have the expected behavior in Ink.

## Example

Same as a link, an image passed to the parser on its own should result in a surrounding paragraph:

`` -> `<p></p>`

And the following Markdown should generate two paragraphs:

```markdown
![cat](cat.jpg)

![dog](dog.jpg)
```

⚡️

```html
<p><img src="cat.jpg" alt="cat"/></p>
<p><img src="dog.jpg" alt="dog"/></p>
```

However, currently Ink outputs:
```html
<img src="cat.jpg" alt="cat"/><img src="dog.jpg" alt="dog"/>
```

## Regression / Implications

Drafting this PR uncovered a different bug which I did not address. The unit tests in `ImageTests.swift` put their references on the next line, like so:

```swift
func testImageWithReference() {
    let html = MarkdownParser().html(from: """
    ![][url]
    [url]: https://swiftbysundell.com
    """)

    XCTAssertEqual(html, #"<p><img src="https://swiftbysundell.com"/></p>"#)
}
```

Meanwhile, the `Link` tests all separate the reference by another line:

```swift
func testLinkWithReference() {
    let html = MarkdownParser().html(from: """
    [Title][url]

    [url]: swiftbysundell.com
    """)

    XCTAssertEqual(html, #"<p><a href="swiftbysundell.com">Title</a></p>"#)
}
```

Testing against other Markdown parsers, either style should work. **However,** currently (and also in this PR) in Ink, the reference **must** be separated by more than just a newline. And by making images behave more like links, those `Image` tests stopped working.

So I introduced newlines in the `Image` tests to separate the references, and they are at least at parity with links now.

## Next steps

I made an effort to fix the reference behavior so that the following test would pass, but couldn't immediately see how to resolve it. @JohnSundell  you may be able to work it out a lot faster!

```swift
func testLinkWithReferenceOnNewline() {
    let html = MarkdownParser().html(from: """
    [Title][url]
    [url]: swiftbysundell.com
    """)

    XCTAssertEqual(html, #"<p><a href="swiftbysundell.com">Title</a></p>"#)
}
```

Thanks again John!

### CODE REVIEW

**Comment by bensyverson** [comment]
Oh, that's good news @john-mueller! I've noticed that some MD parsers do respect a reference link with no blank line, but it's inconsistent. Either way, it's definitely an edge case. Thanks for taking a look at this PR!

**Comment by john-mueller** [comment]
Hey @bensyverson, I'm hoping that @JohnSundell will have some time to merge these soon. I'm thinking it would be good to go ahead and rebase this commit on top of #42 (in other words, rebase your `feature/inline-images` branch on top of your `feature/link-titles` branch. The unit tests from #42 will then need `<p>` tags added in the appropriate places so they keep passing.

I'm thinking it would look something like my [inline-images-fixup](https://github.com/john-mueller/Ink/tree/inline-images-fixup) branch.

After that, maybe just add a note that this PR depends on those changes.
